src/operator/mkl/mkl_batch_norm-inl.h

```diff
@@ -127,11 +127,11 @@ class MKLBatchNormOp : public Operator {
     if (in_data[batchnorm::kData].ndim() == 2) {
       Shape<4> dshape = Shape4(in_data[batchnorm::kData].shape_[0],
                                in_data[batchnorm::kData].shape_[1], 1, 1);
-      data = in_data[batchnorm::kData].get_with_shape<xpu, 4, DType>(dshape, s);
-      out = out_data[batchnorm::kOut].get_with_shape<xpu, 4, DType>(dshape, s);
+      data = in_data[batchnorm::kData].get_with_shape_direct<xpu, 4, DType>(dshape, s);
```

Comment:
please try to do this without touching the interface. Nothing is private in Tblob. You can write a mkl experimental specific wrapper